### DISCUSSION TITLE
Design: tangibility of terrain

### DISCUSSION THREAD

**Discussion by kostmo** [swarm-game/swarm#1538] [Ideas]
As of today, terrain is both:
1. strictly cosmetic, and
2. the sole means to color an "unoccupied" cell.

For (2), this means we have a limited color palette for world aesthetics.
For (1), it means we have no way to affect a robot other than placing an entity.

# Proposal

1. Add a `Custom Text` entry to the terrain sum type
1. Allow scenarios to define their own terrain types and attributes
2. Program special behaviors based on the hard-coded terrain types

**Comment by byorgey** [comment]
Allowing scenarios to define their own custom terrain types and attributes makes a lot of sense, that would be fun.

I'm not sure what you have in mind for "program special behaviors based on the hard-coded terrain types".  How about just adding some kind of sensing command that allows robots to sense the terrain in the current cell?
